src/nccl_ofi_net.c

```diff
@@ -188,34 +189,95 @@ int nccl_net_ofi_create_plugin(nccl_net_ofi_plugin_t **plugin_p)
 			goto exit;
 	}
 
-	/* Select and initialize protocol data structure.
-	 * platform_init() may change the default, so this must occur
-	 * after the platform init call.
+	/* This is ugly, but here's the basic protocol selection
```

Comment:
For this approach, we need to be careful about handling initialization error paths and cleaning things up properly. Although I didn't see anything obviously wrong here.

Reply:
we need to do that anyway!